lib/model/compose.dart

```diff
@@ -0,0 +1,108 @@
+import 'dart:math';
+
+//
+// Put functions for nontrivial message-content generation in this file.
+//
+// If it's complicated enough to need tests, it should go in here.
+//
+
+// https://spec.commonmark.org/0.30/#fenced-code-blocks
+final RegExp _openingBacktickFenceRegex = (() {
+  // Allow "up to three spaces of indentation".
+  // As of Zulip Server 7.0, we don't comply with that detail:
+  //   https://chat.zulip.org/#narrow/stream/6-frontend/topic/quote-and-reply.20fence.20length/near/1588273
+  // and that's a bug.
+  const lineStart = r'^ {0,3}';
```

Comment:
This comment is kind of concerning on the face of it, because it sounds like it's saying that this code's behavior disagrees with the server.

I think just s/and/but/ would already help.

Also helpful would be to be more specific about "we don't comply". I think what's meant is that the server doesn't currently accept those fences. But we're not generating indented fences here, only watching out for them; and it's harmless to make our own fence longer even if the server wouldn't notice the internal fence that we're steering clear of.